ui/src/app/shared/services/workflows-service.ts

```diff
@@ -119,26 +118,11 @@ export class WorkflowsService {
     }
 
     public getContainerLogs(workflow: Workflow, nodeId: string, container: string, archived: boolean): Observable<string> {
-        // we firstly try to get the logs from the API,
-        // but if that fails, then we try and get them from the artifacts
-        const logsFromArtifacts: Observable<string> = Observable.create((observer: Observer<string>) => {
```

Comment:
I don't think we every need to do this. It makes this complex and buggy. Users can just click on the link.

Reply:
additionally, event streams can error under very normal operation

Reply:
IMO even though there is a simple workaround (click the link), I think it's a really nice UX to automatically read logs from the archives if possible.  

I do think we can simplify the logic in here to make it less bug prone though. If it's available, we should look at the logs from the artifact first since if those are present, that means the pod has completed.  If not, fall back to looking at the cluster.  That also means that the event stream is handled last which should cover your comment about how event streams can error under normal conditions.

I'm happy to cook something up later tonight if it'd be helpful, just need to refresh myself on a couple RxJS operators since it's been a few years.